jboss-marshalling/src/main/java/org/infinispan/jboss/marshalling/commons/GenericJBossMarshaller.java

```diff
@@ -27,6 +27,12 @@ public GenericJBossMarshaller(ClassAllowList classAllowList) {
 
    public GenericJBossMarshaller(ClassLoader classLoader, ClassAllowList classAllowList) {
       super();
+      if (classLoader == null) {
+         classLoader = classAllowList != null ? classAllowList.getClassLoader() : null;
```

Comment:
Shouldn't we prioritize the context class loader over the class loader of the class allow list?

Reply:
Not sure :)

Reply:
I worry that the ClassAllowList class will be from a different class loader than the user classes possibly.

Reply:
Ah, ClassAllowList's getClassLoader() is NOT ClassAllowList.class.getClassLoader() :)

Reply:
Aha, my mistake. I was looking at my branch which doesn't have https://github.com/infinispan/infinispan/commit/ad05d66f4e061ea0e4b977b7b591e5a6f9e953b8 in it. It was integrated after I was done.